### PR TITLE
Features/signup

### PR DESCRIPTION
**Pull request by Draqe**
@nzdrml

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
-	resources :posts
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users
+  root 'users#index'
+  get 'users/new' => 'users#new'
```
Comment (nzdrml) [inline config/routes.rb]: If I recall correctly, this is already automatically made available by `resources: users`

Comment (nzdrml) [reply config/routes.rb]: try deleting, including the line below and see if it still works

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_secure_password
+  validates :first_name, :last_name, presence:true
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, presence: true, uniqueness: true
+  validates :password, confirmation: true
+  validates_length_of :password, :in => 6..20, :on => :create
```
Comment (nzdrml) [inline app/models/user.rb]: inconsistent style compared to the rest of the project

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,5 @@
 Rails.application.routes.draw do
-  get 'welcome/index'
   resources :tweets
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  root 'welcome#index'
-
+  resources :users
+ 	root 'users#index'
```
Comment (nzdrml) [inline config/routes.rb]: ...

**Comment by Draqe** [comment]
@nzdrml

**Pull request merged**